src/masoniteorm/query/QueryBuilder.py

```diff
@@ -1006,6 +1006,22 @@ def find_or_fail(self, record_id):
 
         return result
 
+    def find_or_404(self, record_id):
+        """Finds a row by the primary key ID or raise an exception.
+
+        Arguments:
+            record_id {int} -- The ID of the primary key to fetch.
+
+        Returns:
+            Model|HTTP404
+        """
+
+
+        try:
+            return self.find_or_fail(record_id)
+        except ModelNotFound:
+            raise HTTP404()
```

Comment:
@josephmancuso I could repeat the whole code of find_of_fail method, but instead I do that....feeling that it doesn't smell good